### ISSUE TITLE
Web - No Error Generated When Adding a Track Distance Rate of 0.00

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#26344]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to Workspace Settings.
2. Click on "Reimbursements" and proceed to "Track Distance" section.
3. Attempt to add a Track Distance rate of 0.00.

## Expected Result:
A validation error should be presented upon attempting to set a Track Distance rate of 0.00, indicating that the rate cannot be zero.

## Actual Result:
Currently, the system allows the user to save a Track Distance rate of 0.00 without triggering any error messages.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.59.4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/1cdf3280-d11b-4996-8dd1-13eeb3404ac7

https://github.com/Expensify/App/assets/93399543/d2a6e134-4cb7-473a-b46b-17a290dd0622

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Ad-eeb
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692879278812819

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The distance rate field allows entering zero, while it shouldn't

### What is the root cause of that problem?
The Rate field validation does not check the value for zero

### What changes do you think we should make in order to solve the problem?
In the `validate(values)` method, add check for the zero-value:
https://github.com/Expensify/App/blob/a997962e5b67740b3b07f05b09b5a08e5fc1fd0d/src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.js#L97-L99

Will be changed to:
```
if (!rateValueRegex.test(values.rate) || values.rate === '') {
            errors.rate = 'workspace.reimburse.invalidRateError';
        } else if (Number.parseFloat(values.rate) === 0.0) {
            errors.rate = 'workspace.reimburse.zeroRateError'
        }
```

Also, add a corresponding error copy: `workspace.reimburse.zeroRateError`.

<img width="397" alt="image" src="https://github.com/Expensify/App/assets/12595293/e01b4d98-29fa-4b6c-aee6-d4e4fc017771">

### What alternative solutions did you explore? (Optional)

**Comment by pradeepmdk** [comment]
### Proposal
### Please re-state the problem that we are trying to solve in this issue.
The distance rate field allows entering zero, while it shouldn't

### What is the root cause of that problem?

https://github.com/Expensify/App/blob/a997962e5b67740b3b07f05b09b5a08e5fc1fd0d/src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.js#L96C1-L96C1

current regex pattern allowing the 0.0

### What changes do you think we should make in order to solve the problem?
we need update the regex pattern
if any zero digits not allowed means  - `^(?!0+([.]0{1,3})?$)\d{0,8}([.]\d{1,3})?$` 
if we reject only 0.00 means  - `^(?!0[.]00$)\d{0,8}([.]\d{1,3})?$`
if we allowed only zero and not allowed means 0.0, 0.00, 0.000 means `^(?!0[.]0{1,3}?$)\d{0,8}([.]\d{1,3})?$`

**Comment by tamdao** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Web - No Error Generated When Adding a Track Distance Rate of 0.00

### What is the root cause of that problem?
The validation functions only validate the format number and check the value is empty or not
https://github.com/Expensify/App/blob/main/src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.js#L96-L97

### What changes do you think we should make in order to solve the problem?
Update line 97 to add `values.rate <= 0` like this

```js
if (!rateValueRegex.test(values.rate) || values.rate === '' || values.rate <= 0) {
```

**Comment by cubuspl42** [comment]
@tamdao What's the difference between your proposal and [the previous one](https://github.com/Expensify/App/issues/26344#issuecomment-1701373970)?

**Comment by paultsimura** [comment]
@cubuspl42 I don't want to modify my proposal as such, because I'm not sure it won't lose its priority in the queue. But in favor of it, I would say that it's more user-friendly to have as defined validation errors as possible, that's why my proposal contains a separate zero-value validation with a dedicated error message. And this is where my proposal is different from the rest of the proposals.

However, it can be improved by removing the negative-value check from RegExp, and instead checking the number to be strictly positive with the following error: `Rate must be greater than 0`.

This way, the user will have 2 clearly defined validation cases:

1. Incorrect number format – the "Please enter a valid rate" error;
2. Correct number format, but incorrect value (<= 0) – the "Rate must be greater than 0" error.

**Comment by tamdao** [comment]
> @tamdao What's the difference between your proposal and [the previous one](https://github.com/Expensify/App/issues/26344#issuecomment-1701373970)?

Eo. I didn't see. Let I delete my proposal.

**Comment by cubuspl42** [comment]
@tamdao Thank you. In the future, mark such comments as outdated.

**Comment by cubuspl42** [comment]
@hoangzinh 

To sum up, the difference between your proposal and [the first one](https://github.com/Expensify/App/issues/26344#issuecomment-1701373970):
- Do not add extra error message
- Use `<= 0` instead of `=== 0`?

**Comment by cubuspl42** [comment]
The solution by @paultsimura seems straight to the point. I agree that a separate error message feels better, although we'll need to ensure we do this by the book (with adding a translation, etc.).

@hoangzinh The regex already doesn't allow negative values, and otherwise the proposal looks essentially the same.

I say we go with @paultsimura

C+ reviewed 🎀 👀 🎀

**Comment by greg-schroeder** [comment]
Just awaiting @marcochavezf to confirm so we can get started!

**Comment by marcochavezf** [comment]
Thanks for the review @cubuspl42, I agree with the solution and the decision. Assigning @paultsimura 🚀

**Comment by cubuspl42** [comment]
@marcochavezf As we're adding the new copy, should we add "Waiting for copy" label to this issue?

**Comment by paultsimura** [comment]
The PR is ready for the final review – only waiting for the copy approval.

**Comment by marcochavezf** [comment]
Hi @danielrvidal, just a quick copy-check: Is the error message "Rate must be greater than 0" ok?

**Comment by danielrvidal** [comment]
Yes, that works! I'm not sure if we add a punctuation after but I'd use our default method to end the notice.

**Comment by paultsimura** [comment]
@cubuspl42 would it be possible to count the "waiting for copy" as a hold (as per https://github.com/Expensify/App/pull/26745#discussion_r1315773040), or due to the merge freeze we are not eligible for the bonus anyway?

**Comment by greg-schroeder** [comment]
Awaiting deploy to prod

**Comment by cubuspl42** [comment]
- The PR that introduced the bug has been identified. Link to the PR:
    -  It seems that the bug was present in the relevant component since it was created
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
    - N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
    - No need for additional discussion
- Determine if we should create a regression test for this bug.
    - No need for a new regression test
- If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
    - N/A

**Comment by danielrvidal** [comment]
Unassigning myself as I don't think I am needed anymore.

**Comment by paultsimura** [comment]
@marcochavezf @cubuspl42, since @greg-schroeder is OOO until Sep 29, could anybody else proceed with the payment?

**Comment by greg-schroeder** [comment]
Really sorry for the delay, I've was out with an illness. Processing this today.

**Comment by greg-schroeder** [comment]
Issue Participants:

Issue reported by: @Ad-eeb
Contributor: @paultsimura 
C+: @cubuspl42 

Was this issue merged in time to be eligible for the speed bonus? yes
Were their any regressions? no

Payment summary:

Reporter: $50
Contributor: $750
C+: $750

**Comment by greg-schroeder** [comment]
Offer sent to @paultsimura

@Ad-eeb please apply to the job in upwork so I can pay you: https://www.upwork.com/jobs/~01bfe0922db37262d1

**Comment by paultsimura** [comment]
Thanks @greg-schroeder, accepted the offer

**Comment by greg-schroeder** [comment]
Hmm. Thanks. Will pay these now.
